firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/Firestore.kt

```diff
@@ -163,3 +167,35 @@ class FirebaseFirestoreKtxRegistrar : ComponentRegistrar {
     override fun getComponents(): List<Component<*>> =
             listOf(LibraryVersionComponent.create(LIBRARY_NAME, BuildConfig.VERSION_NAME))
 }
+
+/**
+ * Attach a snapshotListener to a DocumentReference and use it as a coroutine flow
+ */
+fun DocumentReference.toFlow() = callbackFlow {
+    val listener = addSnapshotListener { value, error ->
+        if (value != null && value.exists()) {
+            offer(value)
+        } else if (error != null) {
+            close(error)
+        }
+    }
+    awaitClose {
+        listener.remove()
+    }
+}
+
+/**
+ * Attach a snapshotListener to a Query and use it as a coroutine flow
+ */
+fun Query.toFlow() = callbackFlow {
+    val listener = addSnapshotListener { value, error ->
```

Comment:
Same as above, we should allow to pass the `metadataChanges` parameter

Reply:
👍